pages/weekly/[id].tsx

```diff
@@ -0,0 +1,200 @@
+import { marked } from 'marked';
+import { observer } from 'mobx-react';
+import { GetStaticPaths, GetStaticProps } from 'next';
+import { ParsedUrlQuery } from 'querystring';
+import { FC, useContext } from 'react';
+import { Badge, Breadcrumb, Button, Card, Container } from 'react-bootstrap';
+
+import { PageHead } from '../../components/Layout/PageHead';
+import { githubClient } from '../../models/Base';
+import { I18nContext } from '../../models/Translation';
+import styles from '../../styles/Weekly.module.less';
+
+// GitHub Issue type definition
+interface GitHubIssue {
+  id: number;
+  number: number;
+  title: string;
+  body: string | null;
+  state: 'open' | 'closed';
+  labels: Array<{ name: string; color: string } | string>;
+  user: {
+    login: string;
+    avatar_url: string;
+  } | null;
+  created_at: string;
+  updated_at: string;
+  html_url: string;
+}
+
+interface WeeklyDetailParams extends ParsedUrlQuery {
+  id: string;
+}
+
+interface WeeklyDetailProps {
+  issue: GitHubIssue;
+}
+
+export const getStaticPaths: GetStaticPaths<WeeklyDetailParams> = async () => {
+  try {
+    const { body: issues } = await githubClient.get<GitHubIssue[]>(
+      'repos/FreeCodeCamp-Chengdu/IT-Technology-weekly/issues?state=all&sort=created&direction=desc',
+    );
+
+    const paths = (issues || []).map(issue => ({
+      params: { id: issue.number.toString() },
+    }));
+
+    return { paths, fallback: 'blocking' };
+  } catch (error) {
+    console.error('Failed to generate static paths:', error);
+
+    return { paths: [], fallback: 'blocking' };
+  }
+};
+
+export const getStaticProps: GetStaticProps<WeeklyDetailProps, WeeklyDetailParams> = async ({
+  params,
+}) => {
+  const { id } = params!;
+
+  try {
+    const { body: issue } = await githubClient.get<GitHubIssue>(
+      `repos/FreeCodeCamp-Chengdu/IT-Technology-weekly/issues/${id}`,
+    );
+
+    if (!issue) {
+      return {
+        notFound: true,
+      };
+    }
```

Comment:
```suggestion
import { IssueModel } from 'mobx-github';

    const issue = await new IssueModel('FreeCodeCamp-Chengdu', 'IT-Technology-weekly')
        .getOne(id);
```
这个包装方法会将找不到的直接抛错，无需单独处理。